### DISCUSSION TITLE
Testing {cleanepi} and {visdat} on 2024 global.health mpox data

### DISCUSSION THREAD

**Discussion by adamkucharski** [epiverse-trace/epiverse-trace.github.io#316] [General]
I ran cleanepi (cc @Karim-Mane) and [visdat](https://docs.ropensci.org/visdat/) functions on the emerging [global.health mpox dataset](https://reports.global.health/mpox-2024/). Code below. Relatively few entries seemed to be identified as missing, so thing it highlighted was how empty entries are handled if equal to `""`.

Would be interested to hear thoughts on a useful minimal pipeline to recommend on this dataset, as it could be a nice illustrative example of first-pass functionality. 

```r
pak::pak("epiverse-trace/cleanepi")
pak::pak("visdat")
library(cleanepi)
library(visdat)
library(dplyr)

# Load global.health data
data_in <- read.csv("https://mpox-2024.s3.eu-central-1.amazonaws.com/latest.csv")

# Scan with cleanepi
scan_result <- scan_data(data_in)

# Run vis_data on subset
data_in_subset <- data_in |> dplyr::select("Case_status","Age","Gender","Date_onset","Date_confirmation")
vis_dat(data_in_subset)
```

**Comment by Karim-Mane** [comment]
Thank Adam for raising this.

I will work on this and get back to you.

Kind regards,
Karim


From: Adam Kucharski ***@***.***>
Date: Monday, 16 September 2024 at 11:55
To: epiverse-trace/epiverse-trace.github.io ***@***.***>
Cc: Karim MANE ***@***.***>, Mention ***@***.***>
Subject: [Epiverse-TRACE] Testing {cleanepi} and {visdat} on 2024 global.health mpox data (Discussion #316)

I ran cleanepi (cc @Karim-Mane<https://github.com/Karim-Mane>) and visdat<https://docs.ropensci.org/visdat/> functions on the emerging global.health mpox dataset<https://reports.global.health/mpox-2024/>. Code below. Relatively few entries seemed to be identified as missing, so thing it highlighted was how empty entries are handled if equal to "".

Would be interested to hear thoughts on a useful minimal pipeline to recommend on this dataset, as it could be a nice illustrative example of first-pass functionality.

pak::pak("epiverse-trace/cleanepi")

pak::pak("visdat")

library(cleanepi)

library(visdat)

library(dplyr)



# Load global.health data

data_in <- read.csv("https://mpox-2024.s3.eu-central-1.amazonaws.com/latest.csv")



# Scan with cleanepi

scan_result <- scan_data(data_in)



# Run vis_data on subset

data_in_subset <- data_in |> dplyr::select("Case_status","Age","Gender","Date_onset","Date_confirmation")

vis_dat(data_in_subset)

**Comment by spoonerf** [comment]
@adamkucharski in case it is useful here is some [R code](https://gist.github.com/spoonerf/5a76e743d603c6e8e15f003b14165226) that will download the data from the WHO Shiny app which has weekly cases for Africa.

Might be slightly wonky and fall over if they change their Shiny App much, but I couldn't find a more reliable way to download it.
